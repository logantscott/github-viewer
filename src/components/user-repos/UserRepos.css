.UserRepos{
  td {
    border-collapse: collapse;
    border: 1px solid #333;
    padding: 3px;
    margin: 0;
  }
}
