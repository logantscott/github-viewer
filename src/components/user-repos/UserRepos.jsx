import React from 'react';
import PropTypes from 'prop-types';
import styles from './UserRepos.css';

const UserRepos = ({ repos }) => {
  const repositories = repos.map(({ node_id, id, name, description, created_at, updated_at, language, homepage, html_url }) => (
    <tr key={node_id}>
      <td>{id}</td>
      <td><a href={html_url}>{name}</a></td>
      <td>{language}</td>
      <td>{created_at}</td>
      <td>{updated_at}</td>
      <td>{homepage}</td>
      <td>{description}</td>
    </tr>
  ));

  return (
    <>
      <h2 style={{ margin: '10px 0 5px 0' }}>Repositories</h2>
      <table style={{ borderCollapse: 'collapse' }} className={styles.UserRepos}>
        <thead>
          <td>id</td>
          <td>name</td>
          <td>language</td>
          <td>created_at</td>
          <td>udpated_at</td>
          <td>homepage</td>
          <td>description</td>
        </thead>
        {repositories}
      </table>
    </>
  );
};

UserRepos.propTypes = {
  repos: PropTypes.arrayOf(PropTypes.shape({
    id: PropTypes.number.isRequired,
    node_id: PropTypes.number.isRequired,
    name: PropTypes.string.isRequired,
    html_url: PropTypes.string.isRequired,
    description: PropTypes.string,
    created_at: PropTypes.string.isRequired,
    updated_at: PropTypes.string.isRequired,
    homepage: PropTypes.string,
    size: PropTypes.number,
    language: PropTypes.string.isRequired
  }))
};

export default UserRepos;
