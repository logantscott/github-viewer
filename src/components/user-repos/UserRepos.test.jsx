import React from 'react';
import { shallow } from 'enzyme';
import UserRepos from './UserRepos';

const userRepos = [{
  name: 'myRepo',
  html_url: 'someurl',
  id: 1234,
  language: 'english',
  created_at: 'yesterday',
  updated_at: 'today',
  homepage: 'someotherurl',
  description: '',
  node_id: 12345678
}];

describe('UserRepos component', () => {
  it('renders UserRepos', () => {
    const wrapper = shallow(<UserRepos repos={userRepos} />);
    expect(wrapper).toMatchSnapshot();
  });
});
