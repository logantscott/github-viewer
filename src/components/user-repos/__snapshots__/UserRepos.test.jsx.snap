// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserRepos component renders UserRepos 1`] = `
<Fragment>
  <h2
    style={
      Object {
        "margin": "10px 0 5px 0",
      }
    }
  >
    Repositories
  </h2>
  <table
    className="UserRepos"
    style={
      Object {
        "borderCollapse": "collapse",
      }
    }
  >
    <thead>
      <td>
        id
      </td>
      <td>
        name
      </td>
      <td>
        language
      </td>
      <td>
        created_at
      </td>
      <td>
        udpated_at
      </td>
      <td>
        homepage
      </td>
      <td>
        description
      </td>
    </thead>
    <tr
      key="12345678"
    >
      <td>
        1234
      </td>
      <td>
        <a
          href="someurl"
        >
          myRepo
        </a>
      </td>
      <td>
        english
      </td>
      <td>
        yesterday
      </td>
      <td>
        today
      </td>
      <td>
        someotherurl
      </td>
      <td />
    </tr>
  </table>
</Fragment>
`;
